R/snapshot.R

```diff
@@ -791,6 +795,76 @@ renv_snapshot_description_source <- function(dcf) {
 
 }
 
+renv_snapshot_description_infer <- function(dcf) {
+
+  inferred <- tryCatch(
+    renv_snapshot_description_infer_impl(dcf),
+    error = function(err) {
+      warningf("Error inferring package remote: %s", conditionMessage(err))
+      dcf
+    }
+  )
+
+  # if the inferred package version appears to be less than the source one,
+  # don't use it
+  if (renv_version_lt(inferred[["Version"]], dcf[["Version"]]))
+    return(dcf)
+
+  # notify the user if we inferred a different package remote
+  if (!identical(dcf, inferred)) {
+    package <- dcf[["Package"]]
+    remote <- renv_record_format_remote(inferred)
+    writef("* '%s' inferred to have been installed from remote '%s'.", package, remote)
+  }
+
+  inferred
+
+}
+
+renv_snapshot_description_infer_impl <- function(dcf) {
+
+  # if this package appears to have a declared remote, use it
+  for (field in c("RemoteType", "Repository", "biocViews"))
+    if (!is.null(dcf[[field]]))
+      return(dcf)
+
+  # only continue if this appears to be a dev version of a package
+  package <- dcf[["Package"]]
+  version <- unclass(numeric_version(dcf[["Version"]]))[[1]]
```

Comment:
Are you sure you need this block? It would be slightly safer to compare to the CRAN version since not everyone uses the `.9000` convention (including renv 😉), but you're already checking that the inferred version is newer than the source version above.

Reply:
For some reason, I was thinking we'd want to avoid needing to query CRAN here ... but that's probably fine. I'll make that change.